### PR TITLE
improve file read robustness

### PR DESCRIPTION
**Pull request by cnrpman**
Thanks for the great project!

This PR is a simple fix to handle the potential invalid characters in file.
Solve errors like `utf-8' codec can't decode byte 0x92 in position 18`

### CODE REVIEW

File: gpt_index/readers/file.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def load_data(self, **load_kwargs: Any) -> List[Document]:
         data = ""
         data_list = []
         for input_file in self.input_files:
-            with open(input_file, "r") as f:
+            with open(input_file, "r", errors='replace') as f:
```
Comment (jerryjliu) [inline gpt_index/readers/file.py:34]: thanks! could we actually add a param in `__init__` called `errors` that's `replace` by default? then we can just pass through to the `open()` call in `load_data`

Comment (teoh) [inline gpt_index/readers/file.py:34]: I was poking around the docs `errors` and it seems error characters get replaced with a "?" (correct me if i'm wrong)

Have we considered `errors='ignore'` instead?

**Review by teoh — COMMENTED** [review]
thanks for the changes! one small question

**Review by jerryjliu — APPROVED** [review]
thanks!

File: gpt_index/readers/file.py [hunk 2]
```diff
@@ -12,14 +12,20 @@ class SimpleDirectoryReader(BaseReader):
     Can read files into separate documents, or concatenates
     files into one document text.
 
-    input_dir (str): Path to the directory.
-    exclude_hidden (bool): Whether to exclude hidden files (dotfiles).
+    Args:
+        input_dir (str): Path to the directory.
+        exclude_hidden (bool): Whether to exclude hidden files (dotfiles).
+        errors (bool): how encoding and decoding errors are to be handled,
```
Comment (cnrpman) [inline gpt_index/readers/file.py:18]: The `bool` on this line is a typo

Comment (jerryjliu) [reply gpt_index/readers/file.py:18]: oops you're right, i will fix

**Pull request merged**